src/lib/libast/sfio/sfpkrd.c

```diff
@@ -60,170 +107,58 @@ ssize_t sfpkrd(int fd, void *argbuf, size_t n, int rc, long tm, int action) {
     if (rc < 0 && tm < 0 && action <= 0) return sysreadf(fd, buf, n);
 
     t = (action > 0 || rc >= 0) ? (STREAM_PEEK | SOCKET_PEEK) : 0;
-#if !_stream_peek
-    t &= ~STREAM_PEEK;
-#endif
 
     for (ntry = 0; ntry < 2; ++ntry) {
         r = -1;
+
 #if _stream_peek
         if ((t & STREAM_PEEK) && (ntry == 1 || tm < 0)) {
-#ifdef __sun
-            /*
-             * I_PEEK on stdin can hang rsh+ksh on solaris
```

Comment:
I prefer not to fully trust such comments in source code and will leave the code as it is. I won't be surprised if side effects of removing it will show up in some other application than `rsh` on Solaris (or it's derived operating systems).

Reply:
Even if the comment can't be trusted the kludge cannot be trusted. All it is doing is checking if some env vars are unset or set to specific values. Env vars that have nothing to do with rsh per se. This block of code doesn't even rise to the level of "kludge". It's that bad.